diff --git a/telepathy-glib/proxy.c b/telepathy-glib/proxy.c
index f7a5a0efc..639a1f524 100644
--- a/telepathy-glib/proxy.c
+++ b/telepathy-glib/proxy.c
@@ -1885,7 +1885,7 @@ prepare_depends (TpProxy *self,
 /**
  * tp_proxy_prepare_async:
  * @self: an instance of a #TpProxy subclass
- * @features: (transfer none) (array zero-terminated=1) (allow-none): an array
+ * @features: (array zero-terminated=1) (allow-none): an array
  *  of desired features, ending with 0; %NULL is equivalent to an array
  *  containing only 0
  * @callback: if not %NULL, called exactly once, when the features have all
diff --git a/telepathy-glib/simple-client-factory.c b/telepathy-glib/simple-client-factory.c
index 0060b9d0a..a7b5b1ae0 100644
--- a/telepathy-glib/simple-client-factory.c
+++ b/telepathy-glib/simple-client-factory.c
@@ -509,7 +509,7 @@ tp_simple_client_factory_dup_account_features (TpSimpleClientFactory *self,
 /**
  * tp_simple_client_factory_add_account_features:
  * @self: a #TpSimpleClientFactory object
- * @features: (transfer none) (array zero-terminated=1) (allow-none): an array
+ * @features: (array zero-terminated=1) (allow-none): an array
  *  of desired features, ending with 0; %NULL is equivalent to an array
  *  containing only 0
  *
@@ -637,7 +637,7 @@ tp_simple_client_factory_dup_connection_features (TpSimpleClientFactory *self,
 /**
  * tp_simple_client_factory_add_connection_features:
  * @self: a #TpSimpleClientFactory object
- * @features: (transfer none) (array zero-terminated=1) (allow-none): an array
+ * @features: (array zero-terminated=1) (allow-none): an array
  *  of desired features, ending with 0; %NULL is equivalent to an array
  *  containing only 0
  *
@@ -770,7 +770,7 @@ tp_simple_client_factory_dup_channel_features (TpSimpleClientFactory *self,
 /**
  * tp_simple_client_factory_add_channel_features:
  * @self: a #TpSimpleClientFactory object
- * @features: (transfer none) (array zero-terminated=1) (allow-none): an array
+ * @features: (array zero-terminated=1) (allow-none): an array
  *  of desired features, ending with 0; %NULL is equivalent to an array
  *  containing only 0
  *
