diff --git a/telepathy-glib/protocol.c b/telepathy-glib/protocol.c
index 88fdff382..703f79fa1 100644
--- a/telepathy-glib/protocol.c
+++ b/telepathy-glib/protocol.c
@@ -1886,7 +1886,7 @@ _tp_protocol_parse_manager_file (GKeyFile *file,
       TP_PROP_PROTOCOL_PARAMETERS, TP_ARRAY_TYPE_PARAM_SPEC_LIST, param_specs,
       NULL);
 
-  g_ptr_array_unref(param_specs);
+  g_ptr_array_unref (param_specs);
 
   tp_asv_take_boxed (immutables, TP_PROP_PROTOCOL_INTERFACES, G_TYPE_STRV,
       g_key_file_get_string_list (file, group, "Interfaces", NULL, NULL));
