diff --git a/telepathy-glib/introspection.am b/telepathy-glib/introspection.am
index 782064c..b8b84fc 100644
--- a/telepathy-glib/introspection.am
+++ b/telepathy-glib/introspection.am
@@ -14,7 +14,6 @@ INTROSPECTION_FILES = \
 	$(srcdir)/account-manager.c $(srcdir)/account-manager.h \
 	$(srcdir)/automatic-proxy-factory.c $(srcdir)/automatic-proxy-factory.h \
 	$(srcdir)/basic-proxy-factory.c $(srcdir)/basic-proxy-factory.h \
-	$(srcdir)/client-channel-factory.c $(srcdir)/client-channel-factory.h \
 	$(srcdir)/connection.c $(srcdir)/connection.h \
 	$(srcdir)/connection-handles.c $(srcdir)/connection-contact-info.c \
 	$(srcdir)/protocol.c $(srcdir)/protocol.h \
