--- a/tools/c-constants-gen.py
+++ b/tools/c-constants-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 from sys import argv, stdout, stderr
 import xml.dom.minidom
--- a/tools/glib-client-gen.py
+++ b/tools/glib-client-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # glib-client-gen.py: "I Can't Believe It's Not dbus-binding-tool"
 #
--- a/tools/glib-client-marshaller-gen.py
+++ b/tools/glib-client-marshaller-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import sys
 import xml.dom.minidom
--- a/tools/glib-errors-check-gen.py
+++ b/tools/glib-errors-check-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import sys
 import xml.dom.minidom
--- a/tools/glib-errors-str-gen.py
+++ b/tools/glib-errors-str-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import sys
 import xml.dom.minidom
--- a/tools/glib-ginterface-gen.py
+++ b/tools/glib-ginterface-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # glib-ginterface-gen.py: service-side interface generator
 #
--- a/tools/glib-gtypes-generator.py
+++ b/tools/glib-gtypes-generator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Generate GLib GInterfaces from the Telepathy specification.
 # The master copy of this program is in the telepathy-glib repository -
--- a/tools/glib-interfaces-gen.py
+++ b/tools/glib-interfaces-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 from sys import argv, stdout, stderr
 import xml.dom.minidom
--- a/tools/gobject-foo.py
+++ b/tools/gobject-foo.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # gobject-foo.py: generate standard GObject type macros etc.
 #
--- a/tools/make-version-script.py
+++ b/tools/make-version-script.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 """Construct a GNU ld or Debian dpkg version-script from a set of
 RFC822-style symbol lists.
--- a/tools/manager-file.py
+++ b/tools/manager-file.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # manager-file.py: generate .manager files and TpCMParamSpec arrays from the
 # same data (should be suitable for all connection managers that don't have
--- a/tools/xincludator.py
+++ b/tools/xincludator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import sys
 from sys import argv, stdout, stderr
--- a/tests/all-errors-documented.py
+++ b/tests/all-errors-documented.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Check if all the errors have been added to
 # docs/reference/telepathy-glib-sections.txt
 
